main.go

```diff
@@ -53,215 +60,346 @@ func NewClient(login, password string, httpClient *http.Client) *Client {
 	if httpClient == nil {
 		httpClient = &http.Client{
 			Transport: &http.Transport{
-				TLSClientConfig: &tls.Config{InsecureSkipVerify: true},
+				MaxIdleConnsPerHost: 20,
+				TLSClientConfig:     &tls.Config{InsecureSkipVerify: true},
 			},
-			Jar: cookies,
+			Jar:     cookies,
+			Timeout: 1 * time.Minute,
 		}
 	}
 
 	cli := &Client{
-		login:    login,
-		password: password,
-		http:     httpClient,
+		username:   login,
+		password:   password,
+		httpClient: httpClient,
 	}
 
 	return cli
 }
 
-func (cli *Client) Login() error {
-	if cli.uid == "" {
-		uid, err := cli.getUid()
-		if err != nil {
-			return err
-		}
-
-		cli.uid = uid
-	}
-
-	payload := url.Values{
-		"goto":       {urlLoginSuccess},
-		"gotoOnFail": {urlLoginFail},
-		"org":        {"customer"},
-		"old-token":  {cli.login},
-		"IDToken2":   {cli.password},
-		"IDToken1":   {cli.uid},
+func (cli *Client) getDevices() DevicesInfo {
+	req, err := http.NewRequest(http.MethodGet, urlDevices, nil)
+	req.Header.Add("User-Agent", hUserAgent)
+	req.Header.Set("Accept", hContentTypeJson)
+	req.Header.Set("Authorization", basicAuthStr)
+	res, err := cli.httpClient.Do(req)
+	if err != nil {
+		log.Printf("req do err: %s", err)
 	}
+	defer res.Body.Close()
 
-	resp, err := cli.http.PostForm(
-		"https://login.yota.ru/UI/Login",
-		payload,
-	)
-	defer resp.Body.Close()
+	body, _ := ioutil.ReadAll(res.Body)
+	var jsonData map[string]interface{}
+	json.Unmarshal([]byte(body), &jsonData)
```

Comment:
You forgot to check the error here. But I actually think it's something from your debugs where you wanted to check what is coming in the response. Looks like you just forgot to remove these lines.